larq_compute_engine/core/bconv2d/optimized_indirect_bgemm.h

```diff
@@ -25,13 +25,26 @@ inline void BConv2DOptimizedIndirectBGEMM(
 
   const std::int32_t conv_kernel_size =
       bconv2d_params->filter_height * bconv2d_params->filter_width;
-  const std::int32_t bitpacked_input_channels = bitpacked_input_shape.Dims(3);
   const std::int32_t output_size =
       output_shape.Dims(0) * output_shape.Dims(1) * output_shape.Dims(2);
-  const std::int32_t output_channels = bconv2d_params->channels_out;
+  const int32_t groups = bconv2d_params->groups;
+  const int32_t input_depth = bitpacked_input_shape.Dims(3);
+  const int32_t output_channels = bconv2d_params->channels_out;
 
-  indirect_bgemm::RunKernel(kernel, conv_kernel_size, bitpacked_input_channels,
-                            output_size, output_channels, output_transform,
+  // If writing bitpacked output with a channel count that isn't a multiple of
+  // 32 (i.e. where padding bits will be required in the output), fill the
+  // output tensor with zeroes in advance so that the BGEMM doesn't have to
+  // worry about doing the padding.
+  if (std::is_same<DstScalar, TBitpacked>::value &&
+      output_channels % bitpacking_bitwidth != 0) {
```

Comment:
Do we also need to check or assert that `groups == 1` here?

Reply:
I don't think so, I think we still need to do this even for a grouped convolution.